app/models/order_history.rb

```diff
@@ -5,4 +5,16 @@ class OrderHistory < ApplicationRecord
   belongs_to :product
 
   delegate :name, :category_name, :refund_quantity, to: :product, prefix: true
+
+  scope :statistical_detail, (lambda do |month|
+    select("p.name as name_product,
+            SUM(order_histories.quantity) AS product_total_quantity,
+            SUM(order_histories.quantity * order_histories.current_price) AS product_total_price"
+          )
+    .joins(:order)
+    .group("product_id")
+    .joins("INNER JOIN products p ON p.id = product_id")
+    .where(order: { status: Order.statuses[:approved] })
+    .where("EXTRACT(MONTH FROM order.created_at) = ?", month)
```

Comment:
Chỗ này mới where theo tháng thôi nhỉ, vậy năm thì sao?. Ví dụ tháng 2 thì có tháng 2 năm ngoái, và tháng 2 năm nay, ... Như vầy thì có phải sẽ tính toán cho tất cả tháng 2 của mọi năm không. Xử lý thêm điều kiện năm nữa